### PR TITLE
feat(gms): add elasticsearch SSL support

### LINKED ISSUE #1779: Add TLS support for connecting to Elasticsearch
**Is your feature request related to a problem? Please describe.**
[`ESSearchDAO`](https://github.com/linkedin/datahub/blob/master/metadata-dao-impl/elasticsearch-dao/src/main/java/com/linkedin/metadata/dao/search/ESSearchDAO.java) doesn't support https currently. 

**Describe the solution you'd like**
Will most likely need to modify [`RestHighLevelClientFactory`](https://github.com/linkedin/datahub/blob/master/gms/factories/src/main/java/com/linkedin/common/factory/RestHighLevelClientFactory.java) with additional https-related parameters as outlined here: https://www.elastic.co/guide/en/elasticsearch/client/java-rest/current/_encrypted_communication.html

### PR DESCRIPTION
**Pull request by thomasplarsson**
Fixes #1779

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: docker/datahub-gms/env/docker.env [hunk 1]
```diff
@@ -12,3 +12,15 @@ NEO4J_HOST=http://neo4j:7474
...
 NEO4J_USERNAME=neo4j
 NEO4J_PASSWORD=datahub
+
+# Uncomment and set these to support SSL connection to Elasticsearch
+# ELASTICSEARCH_USE_SSL=true
+# ELASTICSEARCH_SSL_PROTOCOL=TLSv1.2
+# ELASTICSEARCH_SSL_KEYSTORE_FILE=
+# ELASTICSEARCH_SSL_TRUSTSTORE_FILE=
+# ELASTICSEARCH_SSL_SECURE_RANDOM_IMPL=
+# ELASTICSEARCH_SSL_TRUSTSTORE_FILE=
```
Comment (jjoyce0510) [inline docker/datahub-gms/env/docker.env:22]: Duplicate?

File: gms/factories/src/main/java/com/linkedin/gms/factory/common/ElasticsearchSSLContextFactory.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    private String sslTrustStoreFile;
+
+    @Value("${ELASTICSEARCH_SSL_TRUSTSTORE_TYPE:#{null}}")
+    private String sslTrustStoreType;
+
+    @Value("${ELASTICSEARCH_SSL_TRUSTSTORE_PASSWORD:#{null}}")
+    private String sslTrustStorePassword;
+
+    @Value("${ELASTICSEARCH_SSL_KEYSTORE_FILE:#{null}}")
+    private String sslKeystoreFile;
```
Comment (jjoyce0510) [inline gms/factories/src/main/java/com/linkedin/gms/factory/common/ElasticsearchSSLContextFactory.java:39]: nit: KeyStore to match the other vars

**Comment by thomasplarsson** [comment]
This is basically a copy of https://github.com/linkedin/datahub/pull/1794 with the only difference being that the creation of the SSLContext was extracted to its own Factory in order to have a better separation of concerns.There is still some duplication in between the two loadRestHttpClient methods but due to how the builder works I didn't find a nice way to share responsibility between them.

The reason for a new PR is the lack of response on the original PR which had some merge conflicts preventing it from being merged.

This has been tested against elasticsearch 6.8 (aws hosted elasticsearch specifically).

File: gms/factories/src/main/java/com/linkedin/gms/factory/common/ElasticsearchSSLContextFactory.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.cert.CertificateException;
+
+
+@Configuration
+public class ElasticsearchSSLContextFactory {
+
+    @Value("${ELASTICSEARCH_SSL_PROTOCOL:#{null}}")
```
Comment (jjoyce0510) [inline gms/factories/src/main/java/com/linkedin/gms/factory/common/ElasticsearchSSLContextFactory.java:23]: Should we provide reasonable defaults for some of these?

File: docker/datahub-gms/env/docker.env (hunk 1 above)
Comment (thomasplarsson) [reply docker/datahub-gms/env/docker.env:20]: Fixed.

File: gms/factories/src/main/java/com/linkedin/gms/factory/common/ElasticsearchSSLContextFactory.java (hunk 2 above)
Comment (thomasplarsson) [reply gms/factories/src/main/java/com/linkedin/gms/factory/common/ElasticsearchSSLContextFactory.java]: Fixed.

File: gms/factories/src/main/java/com/linkedin/gms/factory/common/ElasticsearchSSLContextFactory.java (hunk 3 above)
Comment (thomasplarsson) [reply gms/factories/src/main/java/com/linkedin/gms/factory/common/ElasticsearchSSLContextFactory.java:23]: Not sure. Actual behaviour when not specifying them is fallback to the java platform defaults. Maybe that's good enough?

**Review by shirshanka — APPROVED** [review]
LGTM! Thanks for picking this up @thomasplarsson!

**Pull request merged**